modules/bls-dkg/test/bls-dkg.js

```diff
@@ -8,113 +8,139 @@ const {
+  });
 
-  let userKey
+  let userKey;
 
   before('combines shares into user signing key', function() {
-    userKey = mergeSecretShares([userSecretShares[0], backupSecretShares[0], walletSecretShares[0]])
-  })
+    userKey = mergeSecretShares([userSecretShares[0], backupSecretShares[0], walletSecretShares[0]]);
+  });
 
-  let backupKey
+  let backupKey;
 
   before('combines shares into backup signing key', function() {
-    backupKey = mergeSecretShares([userSecretShares[1], backupSecretShares[1], walletSecretShares[1]])
-  })
+    backupKey = mergeSecretShares([userSecretShares[1], backupSecretShares[1], walletSecretShares[1]]);
+  });
 
-  let walletKey
+  let walletKey;
 
   before('combines shares into wallet signing key', function() {
-    walletKey = mergeSecretShares([userSecretShares[2], backupSecretShares[2], walletSecretShares[2]])
-  })
+    walletKey = mergeSecretShares([userSecretShares[2], backupSecretShares[2], walletSecretShares[2]]);
+  });
 
-  let userSig
+  let userSig;
 
   before('user signs message', async function() {
-    userSig = await sign(message, userKey)
-  })
+    userSig = await sign(message, userKey);
+  });
 
-  let backupSig
+  let backupSig;
 
   before('backup signs message', async function() {
-    backupSig = await sign(message, backupKey)
-  })
+    backupSig = await sign(message, backupKey);
+  });
 
-  let walletSig
+  let walletSig;
 
   before('wallet signs message', async function() {
-    walletSig = await sign(message, walletKey)
-  })
+    walletSig = await sign(message, walletKey);
+  });
 
   it('verifies user-backup sig', async function() {
     const userBackupSig = mergeSignatures({
       1: userSig,
       2: backupSig,
-    })
+    });
     if (!await verify(userBackupSig, message, commonPub)) {
-      throw new Error('Could not verify user + backup signature')
+      throw new Error('Could not verify user + backup signature');
     }
-  })
+  });
 
   it('verifies user-wallet sig', async function() {
     const userWalletSig = mergeSignatures({
       1: userSig,
       3: walletSig,
-    })
+    });
     if (!await verify(userWalletSig, message, commonPub)) {
-      throw new Error('Could not verify user + wallet signature')
+      throw new Error('Could not verify user + wallet signature');
     }
-  })
+  });
 
   it('verifies backup-wallet sig', async function() {
     const backupWalletSig = mergeSignatures({
       2: backupSig,
       3: walletSig,
-    })
+    });
     if (!await verify(backupWalletSig, message, commonPub)) {
-      throw new Error('Could not verify backup + wallet signature')
+      throw new Error('Could not verify backup + wallet signature');
+    }
+  });
+
+  it('verifies user-backup subkey sig with public derived', async function() {
+    const userPublicShareDKG = publicShare(userPolynomial);
+    const derivedUserPublicShareDKG = publicDerive(userPublicShareDKG, subKeyPath, userPolynomial[1]);
```

Comment:
Using `userPolynomial[1]` as the chaincode here is a bit misleading. Please allocate some new entropy to emulate a common chaincode.

Reply:
Sure, forgot to change it. Thanks!